File: app/src/main/java/com/lfork/a98620/lfree/data/source/IMDataSource.java
package com.lfork.a98620.lfree.data.source;

import com.lfork.a98620.lfree.data.DataSource;
import com.lfork.a98620.lfree.data.entity.User;
import com.lfork.a98620.lfree.data.source.remote.imservice.request.LoginListener;

import java.util.List;

public interface IMDataSource extends DataSource {

	/**
	 * 登录成功后，客户端会马上请求用户的详细信息(当然，不包含密码)。 //TODO请求完毕后需要更新TCP本地连接的用户信息
	 */
	void login(User user, LoginListener listener);

	void logout(int userId, GeneralCallback<User> result);

	void getChatUserList(GeneralCallback<List<User>> callback);

	void addChatUser(User user, GeneralCallback<String> callback);

	void removeChatUser(int userId, GeneralCallback<List<User>> callback);

//	void getChatUserInfo(int userId, GeneralCallback<String> callback);

}

File: app/src/main/java/com/lfork/a98620/lfree/data/imdata/IMDataSource.kt
package com.lfork.a98620.lfree.data.imdata

import com.lfork.a98620.lfree.data.DataSource
import com.lfork.a98620.lfree.data.base.entity.User
import com.lfork.a98620.lfree.imservice.request.LoginListener

/**
 *
 * Created by 98620 on 2018/9/3.
 */

interface IMDataSource : DataSource {

    /**
     * 登录成功后，客户端会马上请求用户的详细信息(当然，不包含密码)。 //TODO请求完毕后需要更新TCP本地连接的用户信息
     */
    fun login(user: User, listener: LoginListener)

    fun logout(userId: Int, result: DataSource.GeneralCallback<String>)

    fun getChatUserList(callback: DataSource.GeneralCallback<List<User>>)

    fun addChatUser(user: User, callback: DataSource.GeneralCallback<String>)

    fun addChatUser(user: User, isExisted: Boolean, callback: DataSource.GeneralCallback<String>)

    fun removeChatUser(userId: Int, callback: DataSource.GeneralCallback<List<User>>)

}
